File: 예상대진표.py
def solution(n,a,b):
    cnt = 0
        
    while True : 
        cnt += 1
        if a % 2 == 0 :
            if a-1 == b : 
                break
            a //= 2
        else :
            if a+1 == b : 
                break
                 
            a //= 2
            a += 1

        if b % 2 == 0 :
            b //= 2
        else :
            b //= 2
            print(b)
            b += 1
            print(b)
            
    return cnt;

File: 예상대진표.cpp
##include <iostream>

using namespace std;

int solution(int n, int a, int b){
    int answer = 0;
    int round = 0;
    
    while (a != b) {
        if (a % 2 == 0) {
            a /= 2;
        } else {
            a /= 2;
            a++;
        }

        if (b % 2 == 0) {
            b /= 2;
        } else {
            b /= 2;
            b++;
        }
        
        round++;
    }
    
    cout << a << b << endl;
    answer = round;
    return answer;
}